client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -336,15 +376,25 @@ public void processingFailure(Exception error) {
             state.failConnection(error);
         }
     }
-    
+
     /**
-     * @see SegmentOutputStream#write(java.nio.ByteBuffer,
-     *      java.util.concurrent.CompletableFuture)
+     * @see SegmentOutputStream#write(PendingEvent)
+     *
      */
     @Override
     @Synchronized
-    public void write(PendingEvent event) throws SegmentSealedException {
-        ClientConnection connection = getConnection();
+    @SneakyThrows(SegmentSealedException.class) // We should never encounter SegmentSealedException during a write
```

Comment:
This annotation is not needed. (It has a throw declaration and the exception is on the interface)